enhancements/update/available-update-metadata.md

```diff
@@ -71,15 +73,17 @@ type Release struct {
   // +required
   Image string `json:"image"`
 
-  // URL describes the contents of this release and will include
-  // references to other resources as necessary.  Go here if you know
+  // url describes the contents of this release and will include
+  // references to other resources as necessary. Go here if you know
   // of no more-specific resource addressing your question for this
-  // release image.  For example, this could link errata or other
-  // release notes describing the release image.
+  // release image. This URL is set by the 'url' metadata property on
```

Comment:
`Go here if you know of no more-specific resource addressing your question for this of no more-specific resource addressing your question for this` makes the meaning of the comment odd and confusing. Why do we need to say this here? It does not seem like a necessary sentence because the sentences before it and after it communicates things in a better way.

Reply:
As described [here][1], my resistance to `url`, and preference for `homepage`, revolved around `url` being generic, with no clear semantic meaning.  Clayton's argument was that the semantics *are* that this property references the most-generic resource scoped to this release, i.e. that it is a good choice regardless of which released-scoped information you are seeking.  I want some sort of most-generic statement like my "no more-specific resource" line to get this across.  For example, if you sometimes know of another release-scoped resource (e.g., an entry for this release in a changelog or release notes), falling back to the more-generic `url` when you cannot determine your specific link would be perfectly acceptable, and not a hack.

[1]: https://github.com/openshift/enhancements/pull/123/files#diff-9f5d683af9130c09424e1127af40da1eR197

Reply:
@wking Do you have a concrete example of where this might link to? For instance, for 4.5.2, where would `url` take you?

> For example, if you sometimes know of another release-scoped resource (e.g., an entry for this release in a changelog or release notes), falling back to the more-generic url when you cannot determine your specific link would be perfectly acceptable, and not a hack.

The language is vague enough in the API doc that it's not clear to me when to use something other than `url`. Console always links to the release notes today. Would that sometimes be better?

Reply:
If the concern is that the link won't have detailed release notes, is it enough simply to say that?

Reply:
> Do you have a concrete example of where this might link to?

The most-generic semantics mean that you shouldn't need a concrete example.  Whatever they link to should be generic enough for any release-scoped use case.  But practically, these will continue to link the errata pages for OCP releases, and release-controller detail pages or some such for FCOS releases.

> The language is vague enough in the API doc that it's not clear to me when to use something other than url.

We want the language to clearly communicate:

1. If you know of a specific link for your purpose, by all means, use that.
2. Regardless of whether you know of a specific link for your release-scoped purpose, using the `url` value is fine.  It is the most-generic release-scoped link there is, so it will never be wrong.  The worst it can be is "not as specific as $ALTERNATIVE".

>  Console always links to the release notes today. Would that sometimes be better?

So yes, ignoring this value and linking to the release notes (when you know where they'll be) is fine.  And you should fall back to this value for the "release notes link" use-case when you are not sure of a more-specific release notes target.  We want the API docs to make that clear without folks having to ask for clarification out of band.  Can you float some alternative wording if the current proposal here does not express this clearly?

> If the concern is that the link won't have detailed release notes, is it enough simply to say that?

If detailed release notes exist, the `url` value should either link to them or a more-generic page that in turn links to the release notes (OCP errata link to release notes today).  If detailed release notes exist and the `url` value links to a page from which the release notes are not discoverable, that would be a bug we could file against either ART (who are deciding to link the errata page with `url`) or the Errata folks (who are deciding what content to put on the errata page).

Reply:
> The most-generic semantics mean that you shouldn't need a concrete example. Whatever they link to should be generic enough for any release-scoped use case. But practically, these will continue to link the errata pages for OCP releases, and release-controller detail pages or some such for FCOS releases.

But it's still helpful. Saying something like "For OCP releases, this will typically link to the errata page" makes things immediately clear.

> We want the language to clearly communicate:
> 
>     1. If you know of a specific link for your purpose, by all means, use that.
> 
>     2. Regardless of whether you know of a specific link for your release-scoped purpose, using the `url` value is fine.  It is the most-generic release-scoped link there is, so it will never be wrong.  The worst it can be is "not as specific as $ALTERNATIVE".

I don't know that you need to give clients permission to use another link. People will do that anyway if they want :) And I wouldn't assume the link is ever "wrong." 

IMO, it's more helpful to give examples of what the page is and let clients decide how to use it.

Reply:
> IMO, it's more helpful to give examples of what the page is...

I'm ok with giving examples.  But I think we're doing it wrong if the only way we can describe the intended values is by saying "it's going to be content like $existing_examples".  We should be able to describe the content in a way that stands alone, and then supplement that normative definition with informative examples.

> I don't know that you need to give clients permission to use another link. People will do that anyway if they want :) And I wouldn't assume the link is ever "wrong."

If we had URI properties for `license` and `releaseNotes`, using the value of `releaseNotes` and saying "license is here" when `license` was unset would be clearly wrong.  The special thing about the most-generic `url` is that saying "license is here" and linking the the value of `url` would be perfectly fine, because you could assume that the license information would be easily discoverable from the generic `url` target.

Reply:
Restored:

> For example, this could link errata or other release notes describing the release image.

with b22d4e6 -> 73a1139, to address @spadgett's "[Do you have a concrete example of where this might link to?][1]".

[1]: https://github.com/openshift/enhancements/pull/408#discussion_r459713353

Reply:
I don't like this language.  Errata is confusing to non OCP customers, release notes is confusing to OCP customers.  This sentence is not a good sentence for those reasons.

> The URL will contain information about the release.

is the minimum bar.  I am fine with leaving OCP or OKD language out right now.

Reply:
> I am fine with leaving OCP or OKD language out right now.

Ok, back to b22d4e6.

Reply:
> The URL will contain information about the release.

Pushed b22d4e6 -> d859e06, replacing the first sentence with Clayton's new suggestion.